public/components/common/custom-search-bar/custom-search-bar.tsx

```diff
@@ -1,165 +1,182 @@
-import React, { useState, useEffect, useLayoutEffect } from 'react';
+import React, { useState, useEffect } from 'react';
 
-import { getIndexPattern } from '../../overview/mitre/lib'
 import { Filter } from '../../../../../../src/plugins/data/public/';
-import { FilterMeta, FilterState, FilterStateStore } from '../../../../../../src/plugins/data/common';
-
 import {
-    EuiFlexGroup,
-    EuiFlexItem,
-    EuiSwitch,
-} from '@elastic/eui';
+  FilterMeta,
+  FilterState,
+  FilterStateStore,
+} from '../../../../../../src/plugins/data/common';
+import { AppState } from '../../../react-services/app-state';
+
+import { EuiFlexGroup, EuiFlexItem, EuiSwitch } from '@elastic/eui';
 
 //@ts-ignore
-import { getDataPlugin } from '../../../kibana-services';
 import { KbnSearchBar } from '../../kbn-search-bar';
-import { Combobox } from './components'
+import { Combobox } from './components';
+import { useFilterManager, usePrevious } from '../hooks';
 
 export const CustomSearchBar = ({ filtersValues, ...props }) => {
-
-    const KibanaServices = getDataPlugin().query;
-    const filterManager = KibanaServices.filterManager;
-    const indexPattern = getIndexPattern();
-    const defaultSelectedOptions = () => {
-        const array = []
-        filtersValues.forEach(item => {
-            array[item.key] = []
-        })
-
-        return array
-    }
-    const [avancedFiltersState, setAvancedFiltersState] = useState(false);
-    const [selectedOptions, setSelectedOptions] = useState(defaultSelectedOptions);
-
-    useEffect(() => {
-        let filterSubscriber = filterManager.getUpdates$().subscribe(() => {
-            onFiltersUpdated()
-            return () => {
-                filterSubscriber.unsubscribe();
-            };
-        });
-    }, []);
-
-    const onFiltersUpdated = () => {
-        refreshCustomSelectedFilter()
+  const { filterManager, filters } = useFilterManager();
+  const defaultSelectedOptions = () => {
+    const array = [];
+    filtersValues.forEach((item) => {
+      array[item.key] = [];
+    });
+
+    return array;
+  };
+  const [avancedFiltersState, setAvancedFiltersState] = useState(false);
+  const [selectedOptions, setSelectedOptions] = useState(defaultSelectedOptions);
```

Comment:
Over the code, setSelectedOptions is used to set references to functions rather than running the functions and setting the state as their return, but the variable selectedOptions is used with the structure of an array.